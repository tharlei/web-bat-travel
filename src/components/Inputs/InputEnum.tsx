import { BsChevronDown } from 'react-icons/bs';

export interface OptionData {
  value: string;
  label: string;
}

interface InputEnumProps {
  label: string;
  setValue: (value: string) => void;
  options: OptionData[];
}

export function InputEnum({ label, options, setValue }: InputEnumProps) {
  return (
    <div className="flex relative items-center">
      <div className="flex flex-col w-full">
        <label className="uppercase text-gray-900 text-xs -mb-6 ml-2 z-10">
          {label}
        </label>
        <select
          required
          onChange={(event) => setValue(event.target.value)}
          className="bg-gray-100 rounded border-0 pt-8"
        >
          <option></option>
          {options.map((option) => (
            <option key={option.value} value={option.value}>
              {option.label}
            </option>
          ))}
        </select>
      </div>
      <BsChevronDown className="text-cyan text-2xl h-full absolute font-bold right-3 bg-gray-100 z-20" />
    </div>
  );
}
