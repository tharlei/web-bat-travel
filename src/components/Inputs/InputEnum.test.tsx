import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { InputEnum } from './InputEnum';

describe('InputEnum component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <InputEnum setValue={(value) => value} label="Test" options={[]} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
