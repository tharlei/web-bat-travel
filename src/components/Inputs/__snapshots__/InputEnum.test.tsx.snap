// Vitest Snapshot v1

exports[`InputEnum component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex relative items-center"
  >
    <div
      class="flex flex-col w-full"
    >
      <label
        class="uppercase text-gray-900 text-xs -mb-6 ml-2 z-10"
      >
        Test
      </label>
      <select
        class="bg-gray-100 rounded border-0 pt-8"
        required=""
      >
        <option />
      </select>
    </div>
    <svg
      class="text-cyan text-2xl h-full absolute font-bold right-3 bg-gray-100 z-20"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</DocumentFragment>
`;
